src/hardware/vga_draw.cpp

```diff
@@ -1198,6 +1199,39 @@ static void maybe_aspect_correct_tall_modes(double &current_ratio)
 	current_ratio *= 2;
 }
 
+// Pre-VGA color modes often ran with a horizonal sync rate of 15.7 kHz
+// (200-lines) or 21.85 KHz (350-lines), both producing a vertical sync rate of
+// 60 Hz on those old systems, however newer VGA systems often run at a slightly
+// higher clock rate. So we adjust the clock when not running a pre-VGA system.
+// Ref: http://minuszerodegrees.net/video/bios_video_modes.htm
+static uint32_t get_clock_bias_for_pre_vga_color_mode(const uint32_t clock_hz,
+                                                      const uint32_t htotal,
+                                                      const uint32_t vtotal)
+{
+	constexpr uint32_t pre_vga_color_modes = (M_EGA | M_CGA2 |
+	                                          M_CGA2_COMPOSITE | M_CGA4 |
+	                                          M_CGA4_COMPOSITE | M_CGA16 |
+	                                          M_TANDY2 | M_TANDY4 | M_TANDY16);
+
+	const auto is_pre_vga_color_mode = bit::any(static_cast<uint32_t>(vga.mode),
+	                                            pre_vga_color_modes);
+
+	const auto is_200_of_350_line_mode = (CurMode->sheight == 200 ||
```

Comment:
Shouldn't it be "or" instead of "of"?

Reply:
Alternatively, you could call it "is_lowres_mode" (but that's a bit ambiguous perhaps).

Reply:
``` text
  | = H.SYNC: positive at 15.7 kHz, V.SYNC: positive at 60 Hz       (A.K.A. EGA Mode 1)
-- | --
  | = H.SYNC: positive at 21.85 kHz, V.SYNC: negative at 60 Hz    (A.K.A. EGA Mode 2 or EGA High Resolution)
```

They mention 'mode 2 (350 lines) as "EGA High Res"; but yeah - given we're comparing all of the possible modes (even beyond EGA) in this function, low-res applies.  Will do :+1: